samples/snippets/csharp/new-in-6/string-interpolation.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+
+namespace StringInterpolationTutorial
+{
+    public class Examples
+    {
+        public void StringFormatExample()
+        {
+            // <StringFormatExample>
+            var firstName = "Matt";
+            var lastName = "Groves";
+            var str = String.Format("My name is {FirstName} {LastName}", firstName, lastName);
+            Console.WriteLine(str);
+            // </StringFormatExample>
+        }
+
+        public void InterpolationExample()
+        {
+            // <InterpolationExample>
+            var firstName = "Matt";
+            var lastName = "Groves";
+            var str = $"My name is {firstName} {lastName}");
+            Console.WriteLine(str);
+            // </InterpolationExample>
+        }
+
+        public void InterpolationExpressionExample()
+        {
+            // <InterpolationExpressionExample>
+            for(var i = 0; i < 5; i++) {
+                Console.WriteLine($"This is line number {i + 1}");
+            }
+            // </InterpolationExpressionExample>
+        }
+
+        public void InterpolationFormattingExample()
+        {
+            // <InterpolationFormattingExample>
+            var rand = new Random();
+            for(var i = 998; i < 1005; i++)
+            {
+                var randomDecimal = rand.NextDouble() * 10000;
+                Console.WriteLine($"{i, -10} {randomDecimal, 6:N2}");
+            }
+            // </InterpolationFormattingExample>
+        }
+
+        public void InterpolationInternationalizationExample()
+        {
+            // <InterpolationInternationalizationExample>
+            var birthday = new DateTime(1980, 1, 29);
+            Console.WriteLine($"My birthday is {birthday}");
+            // This outputs "My birthday is 1/29/1980 12:00:00 AM"
+
+            var birthdayFormattable = (IFormattable)$"My birthday is {birthday}";
+            Console.WriteLine(birthdayFormattable.ToString(null, new CultureInfo("fr-FR")));
+            // This outputs "My birthday is 29/01/1980 00:00:00"
+            // </InterpolationInternationalizationExample>
+        }
+
+        public void InterpolationLocalizationExample()
+        {
+            // <InterpolationLocalizationExample>
+            var animal = "fox";
+            var localizeMe = $"The {adj} brown {animal} jumped over the lazy {otheranimal}";
```

Comment:
Because these won't compile, you should put them inline in the markdown file, using the ``` syntax. Otherwise, the CI build will fail.